INDI/INDI/indiserver.c

```diff
@@ -1119,109 +1113,152 @@ static void newFIFO(void)
+            }
+            if (!desc) { continue; }
+            strncpy(targ, var[j], MAXSBUF);
+            targ[MAXSBUF - 1] = '\0';
+            if (verbose)
+            {
+                fprintf(stderr, "%s: With %s: [%s]\n", ts, desc, targ);
+            }
+#           if 0
             if (arg[j][0] == 'n')
             {
-                strncpy(tName, var[j], MAXSBUF - 1);
+                strncpy(tName, var[j], MAXSBUF);
                 tName[MAXSBUF - 1] = '\0';
 
                 if (verbose)
+                {
                     fprintf(stderr, "%s: With name: %s\n", ts, tName);
+                }
             }
             else if (arg[j][0] == 'c')
             {
-                strncpy(envConfig, var[j], MAXSBUF - 1);
+                strncpy(envConfig, var[j], MAXSBUF);
                 envConfig[MAXSBUF - 1] = '\0';
 
                 if (verbose)
+                {
                     fprintf(stderr, "%s: With config: %s\n", ts, envConfig);
+                }
             }
             else if (arg[j][0] == 's')
             {
-                strncpy(envSkel, var[j], MAXSBUF - 1);
+                strncpy(envSkel, var[j], MAXSBUF);
                 envSkel[MAXSBUF - 1] = '\0';
 
                 if (verbose)
-                    fprintf(stderr, "%s: With skeketon: %s\n", ts, envSkel);
+                {
+                    fprintf(stderr, "%s: With skeleton: %s\n", ts, envSkel);
+                }
             }
             else if (arg[j][0] == 'p')
             {
-                strncpy(envPrefix, var[j], MAXSBUF - 1);
+                strncpy(envPrefix, var[j], MAXSBUF);
                 envPrefix[MAXSBUF - 1] = '\0';
 
                 if (verbose)
+                {
                     fprintf(stderr, "%s: With prefix: %s\n", ts, envPrefix);
+                }
             }
+#           endif //0
         }
 
+#       if 0
         if (!strcmp(cmd, "start"))
             startCmd = 1;
         else
             startCmd = 0;
+#       endif //0
+        startCmd = !strcmp(cmd, "start") ? 1 : 0;
 
         if (startCmd)
         {
             if (verbose)
+            {
                 fprintf(stderr, "%s: FIFO: Starting driver %s\n", ts, tDriver);
-            dp = allocDvr();
-            strncpy(dp->name, tDriver, MAXINDIDEVICE);
+            }
+
+            if ((dp=findActiveDvrInfo(tDriver)))
+            {
+                if (!*findDvrInRestartList(dp))
+                {
+                    if (verbose)
+                    {
+                        fprintf(stderr, "%s: FIFO: Skipping driver %s that is already started\n", ts, tDriver);
+                    }
+                    continue;
+                }
+                removeDvrFromRestartList(dp);
+            }
+
+            if (!dp)
+            {
+                dp = allocDvr();
+                strncpy(dp->name, tDriver, MAXINDINAME);
+                dp->name[MAXINDINAME-1] = '\0';
+            }
```

Comment:
Again, this change ensures a start command does not duplicate an existing driver.